crates/league-mod/src/utils/league_path.rs

```diff
@@ -0,0 +1,295 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+/// Validates if a path points to a valid League of Legends executable
+/// The path must exist and point to "League of Legends.exe"
+pub fn is_valid_league_path(path: &str) -> bool {
+    let p = Path::new(path);
+
+    // Check if file exists
+    if !p.exists() {
+        return false;
+    }
+
+    // Check if it's the correct executable name
+    if let Some(file_name) = p.file_name() {
+        if let Some(name_str) = file_name.to_str() {
+            return name_str == "League of Legends.exe";
+        }
+    }
+
+    false
+}
+
+/// Get all available drive letters on Windows using WMIC
+/// Returns common drives if detection fails
+fn get_available_drives() -> Vec<String> {
```

Comment:
I suggest using the `sysinfo` crate here as well. You can get the available drives from here:
https://docs.rs/sysinfo/latest/sysinfo/struct.Disks.html#method.new_with_refreshed_list